fhir2dataset/parser.py

```diff
@@ -0,0 +1,193 @@
+        if len(item_parsed) == 2:
+            resource_type = item_parsed[0]
+            alias = item_parsed[1]
+            self.__from[alias] = resource_type
+        elif len(item_parsed) == 1:
+            resource_type = item_parsed[0]
+            alias = item_parsed[0]
+            self.__from[alias] = resource_type
+        else:
+            raise ValueError
+        # print(f"self.__from: {self.__from}")
+
+    def __inner_join_parser(self, string):
+        alias_parent, searchparam_parent, alias_child = self.__join_parser(string)
+        self.__inner_join[alias_parent][searchparam_parent] = alias_child
+
+    def __child_join_parser(self, string):
+        alias_parent, searchparam_parent, alias_child = self.__join_parser(string)
+        self.__child_join[alias_parent][searchparam_parent] = alias_child
+
+    def __parent_join_parser(self, string):
+        alias_parent, searchparam_parent, alias_child = self.__join_parser(string)
+        self.__parent_join[alias_parent][searchparam_parent] = alias_child
+
+    def __join_parser(self, string):
+        item_parsed = re.split(self.__split_mask_from_join, string)
+        # print("\nJOIN")
+        # print(f"item_parsed: {item_parsed}")
+        assert len(item_parsed) == 2
+        self.__from_parser(item_parsed[0])
+
+        join_conditions = re.split(self.__split_mask_join, item_parsed[1])
+        assert len(join_conditions) == 2
+
+        alias_parent = None
+        searchparam_parent = None
+        alias_child = None
+        for join_condition in join_conditions:
+            alias, join_rule = re.split(r"\.", join_condition, 1)
+            assert alias in self.__from.keys(), (
+                f"The {alias} alias used in the {join_condition} join must have been referenced "
+                "behind a FROM, INNER JOIN, PARENT JOIN or CHILD JOIN. "
+                f"The only aliases referenced are here: {self.__from}"
+            )
+            if join_rule in ["id", "id ", " id", " id "]:
+                if alias_child:
+                    raise ValueError
+                alias_child = alias
+            else:
+                if alias_parent or searchparam_parent:
+                    raise ValueError
+                alias_parent = alias
+                searchparam_parent = join_rule
+
+        return alias_parent, searchparam_parent, alias_child
+
+    def __where_parser(self, string):
+        item_parsed = re.split(self.__split_mask_where, string)
+        # print("\nWHERE")
+        # print(f"item_parsed: {item_parsed}")
+        for where_condition in item_parsed:
+            condition_parsed = re.split(self.__split_mask_where_condition, where_condition)
+            assert len(condition_parsed) == 3
+            alias, where_rule = re.split(r"\.", condition_parsed[0], 1)
+            value = condition_parsed[2]
+            if condition_parsed[1] in PREFIX:
+                prefix = condition_parsed[1].replace(" ", "")
```

Comment:
can you use trim() instead?

Reply:
I replaced it with '.strip()', is that what you were referring to?